internal/servers/worker/internal/metric/controller_dialer.go

```diff
@@ -0,0 +1,156 @@
+package metric
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/globals"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+const (
+	labelGrpcCode               = "grpc_code"
+	labelGrpcService            = "grpc_service"
+	labelGrpcMethod             = "grpc_method"
+	workerToControllerSubsystem = "worker_to_controller_cluster"
+
+	controllerServersPackage = "controller.servers.services.v1"
+)
+
+// grpcRequestLatency collects measurements of how long a round-trip
+// gRPC request between a worker and controller cluster takes.
+var grpcRequestLatency prometheus.ObserverVec = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Namespace: globals.MetricNamespace,
+		Subsystem: workerToControllerSubsystem,
+		Name:      "grpc_request_duration_seconds",
+		Help:      "Histogram of latencies for cluster gRPC requests between a worker and controller.",
+		Buckets:   prometheus.DefBuckets,
+	},
+	[]string{labelGrpcCode, labelGrpcService, labelGrpcMethod},
+)
+
+// statusFromError retrieves the *status.Status from the provided error.  It'll
+// attempt to unwrap the *status.Error, which is something status.FromError
+// does not do.
+func statusFromError(err error) *status.Status {
+	if s, ok := status.FromError(err); ok {
+		return s
+	}
+
+	type gRPCStatus interface {
+		GRPCStatus() *status.Status
+	}
+	var unwrappedStatus gRPCStatus
+	if ok := errors.As(err, &unwrappedStatus); ok {
+		return unwrappedStatus.GRPCStatus()
+	}
+
+	return status.New(codes.Unknown, "Unknown Code")
+}
+
+func splitMethodName(fullMethodName string) (string, string) {
+	fullMethodName = strings.TrimPrefix(fullMethodName, "/") // remove leading slash
+	if i := strings.Index(fullMethodName, "/"); i >= 0 {
+		return fullMethodName[:i], fullMethodName[i+1:]
+	}
+	return "unknown", "unknown"
+}
+
+type requestRecorder struct {
+	labels prometheus.Labels
+
+	// measurements
+	start time.Time
+}
+
+func newRequestRecorder(req interface{}, fullMethodName string) requestRecorder {
+	service, method := splitMethodName(fullMethodName)
+	r := requestRecorder{
+		labels: prometheus.Labels{
+			labelGrpcMethod:  method,
+			labelGrpcService: service,
+		},
+		start: time.Now(),
+	}
+
+	return r
+}
+
+func (r requestRecorder) record(resp interface{}, err error) {
+	r.labels[labelGrpcCode] = statusFromError(err).Code().String()
+	grpcRequestLatency.With(r.labels).Observe(time.Since(r.start).Seconds())
+}
+
+// InstrumentControllerDialer wraps a UnaryClientInterceptor and records
+// observations for the collectors associated with the worker<->controller
+// gRPC connection.
+func InstrumentControllerDialer() grpc.UnaryClientInterceptor {
```

Comment:
This isn't a dialer.  This is passed in as a dialer option, but is actually an interceptor.  Consider renaming this function to both specify that this is for the cluster (or server serivce) client and it's an interceptor, not a dialer.

Here and elsewhere you refer to this as dialer instrumentation.

Reply:
Addressed [072e97e](https://github.com/hashicorp/boundary/pull/1998/commits/072e97e4eeddfe56f09d75d23e49760605c3b21a)